services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx

```diff
@@ -127,6 +131,39 @@ export const SubmissionSummary = (): React.ReactElement => {
     const modalFormInitialValues = {
         unlockReason: '',
     }
+
+    // document date lookup state
+    const [documentDates, setDocumentDates] = useState<DocumentDateLookupTable>(
+        {}
+    )
+
+    const makeDateTable = (submissions: Submission2) => {
+        const docBuckets = [
+            'contractDocuments',
+            'rateDocuments',
+            'documents',
+        ] as const
+        const lookupTable = {} as DocumentDateLookupTable
+        if (submissions) {
+            submissions.revisions.forEach((revision) => {
+                const revisionData = base64ToDomain(
+                    revision.revision.submissionData
+                )
+                if (revisionData instanceof Error) {
+                    console.error(
+                        'failed to read submission data; unable to display document dates'
+                    )
+                    return
+                }
+                docBuckets.forEach((bucket) => {
+                    revisionData[bucket].forEach((doc) => {
+                        lookupTable[doc.name] = revisionData.updatedAt
```

Comment:
Do we need to protect against unique keys at all here? I'm hoping not. What if the document with same name is added/remove in different revisions - we just keep the most recent update date right?

Reply:
I made sure our upload dupe check was relying on name before I used it here, so I think this should be fine.